src/Error/ExceptionRenderer.php

```diff
@@ -390,29 +390,37 @@ protected function _outputMessage(string $template): Response
             $this->controller->render($template);
 
             return $this->_shutdown();
+        } catch (MissingLayoutException | MissingPluginException $e) {
+            // Configuration error, bail out of everything
         } catch (MissingTemplateException $e) {
-            $attributes = $e->getAttributes();
-            if (
-                $e instanceof MissingLayoutException ||
-                (
-                    isset($attributes['file']) &&
-                    strpos($attributes['file'], 'error500') !== false
-                )
-            ) {
-                return $this->_outputMessageSafe('error500');
+            $request = $this->controller->getRequest();
+
+            // Bail out of one configuration at a time trying to find template
+            if ($request->getParam('_ext')) {
+                $this->controller->setRequest($request->withParam('_ext', null));
+
+                return $this->_outputMessage($template);
             }
 
-            return $this->_outputMessage('error500');
-        } catch (MissingPluginException $e) {
-            $attributes = $e->getAttributes();
-            if (isset($attributes['plugin']) && $attributes['plugin'] === $this->controller->getPlugin()) {
+            if ($request->getParam('prefix')) {
+                $this->controller->setRequest($request->withParam('prefix', null));
+
+                return $this->_outputMessage($template);
+            }
+
+            if ($this->controller->getPlugin()) {
```

Comment:
We could skip this check. Having the controller's plugin property not match the one in the request should be discouraged IMO. I would even like to deprecate this method.

Reply:
This is an awkward one. I wasn't sure what to when rendering something like a pdf from a plugin which doesn't have a pdf-specific error template. Do we check for plugin-specific error template or jump back to the application templates immediately?

Reply:
I doubt anyone is going to generate an actual pdf for an error page.

Reply:
Right. The purpose of this PDF allowing the non-pdf error template to render.